### PR TITLE
Feat: Add a delete job command and the corresponding api

### PR DESCRIPTION
**Pull request by islavallee**
Add ta call to the kuberenetes api to be able to delete a job in a namespace
Refactor the delete namespace command and create new delete job command:
 - blackbeard delete namespace
 - blackbeard delete job

### CODE REVIEW

File: cmd/delete.go [hunk 1]
```diff
@@ -1,54 +1,47 @@
...
+// deleteCmd represents the delete command
 var deleteCmd = &cobra.Command{
-	Use:   "delete",
-	Short: "Delete a namespace",
-	Long:  `This command delete a namespace and all the associated resources`,
+	Use:   "delete [command]",
+	Short: "Delete an object",
+	Long: `Delete resources by namespace or names.
 
+Deletetion of a namespace will remove the namespace all his attached object and the intentory attached to it. While the remove of the resource will only supress it form the namespace.`,
```
Comment (pboiseau) [inline cmd/delete.go:18]: Deletetion of a namespace will delete the namespace and remove all his attached object with the intentory attached to it. While removing an object will only supress it from the namespace but keep everything else.`,

File: cmd/delete.go [hunk 2]
```diff
@@ -1,54 +1,47 @@
 package cmd
 
 import (
-	"errors"
-	"os"
-
+	"bytes"
 	"fmt"
+	"html/template"
```
Comment (pboiseau) [inline cmd/delete.go:6]: maybe try with `text/template` as we are not using any html in prompt

File: cmd/delete.go [hunk 3]
```diff
@@ -1,54 +1,47 @@
...
+	data := []string{"delete namespace", "delete job"}
 
-	api := newAPI(newFileClient(dir), newKubernetesClient())
-
-	err := api.Delete(namespace, false)
-	if err != nil {
-		return err
+	contents := bytes.Buffer{}
+	if err := tpl.Execute(&contents, data); err != nil {
+		logrus.Fatalf("error when executing template : %v", err)
```
Comment (pboiseau) [inline cmd/delete.go:43]: error while executing template

File: pkg/http/resource_handler.go [hunk 4]
```diff
@@ -80,3 +80,17 @@ func (h *Handler) GetStatuses(c *gin.Context) {
...
 }
+
+// Delete handle the namespace deletion.
+func (h *Handler) DeleteResource(c *gin.Context) {
+	namespace := c.Params.ByName("namespace")
+	resource := c.Params.ByName("resource")
+
+	//Delete inventory
+	if err := h.api.DeleteResource(namespace, resource); err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
```
Comment (pboiseau) [inline pkg/http/resource_handler.go:91]: I think return a http 500 error code is not a good idea. Maybe return a http.StatusBadRequest instead

File: swagger.json [hunk 5]
```diff
@@ -366,6 +366,43 @@
...
+          }
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "204": {
+            "description": "Deletion successfully executed"
+          },
+          "500": {
```
Comment (pboiseau) [inline swagger.json:397]: maybe return a 400 instead

File: cmd/delete_job.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var deleteJobCmd = &cobra.Command{
+	Use:   "job [NAME]",
+	Short: "Delete a job object from a namespace",
+	Long: `Delete a job object that started a pod.
+
+It won't remove any configuration in the inventory. Reappling the inventory will redeploy it.
```
Comment (pboiseau) [inline cmd/delete_job.go:16]: Reapplying

File: swagger.json [hunk 7]
```diff
@@ -366,6 +366,43 @@
         }
```
Comment (pboiseau) [inline swagger.json:366]: I think you can also update docs/workflow/cli.md

File: pkg/http/resource_handler.go (hunk 4 above)
Comment (islavallee) [reply pkg/http/resource_handler.go:91]: The error could be any of those. If a user ask to delete an unknown resource it should be a http.StatusBadRequest. But if there was an issue with the kubernetes api then i think a http.StatusInternalServerError is more suited.

**Pull request merged**